cmd/satellite/reports/attribution.go

```diff
@@ -0,0 +1,98 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package reports
+
+import (
+	"context"
+	"encoding/csv"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/skyrings/skyring-common/tools/uuid"
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+
+	"storj.io/storj/internal/memory"
+	"storj.io/storj/satellite/attribution"
+	"storj.io/storj/satellite/satellitedb"
+)
+
+// GenerateAttributionCSV creates a report with
+func GenerateAttributionCSV(ctx context.Context, database string, partnerID uuid.UUID, start time.Time, end time.Time, output io.Writer) error {
+	db, err := satellitedb.New(zap.L().Named("db"), database)
+	if err != nil {
+		return errs.New("error connecting to master database on satellite: %+v", err)
+	}
+	defer func() {
+		err = errs.Combine(err, db.Close())
+	}()
+
+	rows, err := db.Attribution().QueryAttribution(ctx, partnerID, start, end)
+	if err != nil {
+		return err
+	}
+
+	w := csv.NewWriter(output)
+	headers := []string{
+		"projectID",
+		"bucketName",
+		"byte-hours:Remote",
+		"byte-hours:Inline",
+		"bytes:BWEgress",
+	}
+	if err := w.Write(headers); err != nil {
+		return err
+	}
+
+	for _, row := range rows {
+		record, err := csvRowToStringSlice(row)
+		if err != nil {
+			return err
+		}
+		if err := w.Write(record); err != nil {
+			return err
+		}
+	}
+	if err := w.Error(); err != nil {
+		return err
+	}
+	w.Flush()
```

Comment:
we should defer this right after `csv.NewWriter(output)`

Reply:
Updated